src/trace-server.ts

```diff
@@ -43,6 +49,7 @@ export class TraceServer {
             let id: NodeJS.Timeout;
             // recovering from workspaceState => no this.server set
             if (this.server) {
+                this.server.removeAllListeners();
```

Comment:
Using `this.server.removeAllListeners()` would also remove any listeners registered by node.js' `child_process.js`, which could have undesirable side-effects. It would be better to use explicitly defined handlers for the listeners we register, and use `this.server.off('<event>', <handler>)` to remove them when no longer needed. Optionally, maybe a "disposable" pattern could be used?

Reply:
Make sense. I'll update that.